app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -1416,4 +1421,177 @@ class PatientRepositoryAndroidTest {
     assertThat(patientRepository.findPatientWithBusinessId(deletedBusinessIdentifier).blockingFirst()).isEqualTo(None)
     assertThat(patientRepository.findPatientWithBusinessId("missing_identifier").blockingFirst()).isEqualTo(None)
   }
+  
+  @Test
+  fun if_patient_is_defaulter_then_isPatientDefaulter_should_return_true() {
+
+    val defaulterPatients = arrayListOf<String>()
+
+    fun savePatientRecord(
+        fullName: String,
+        bpMeasurement: List<BloodPressureMeasurement>?,
+        hasHadHeartAttack: MedicalHistory.Answer = NO,
+        hasHadStroke: MedicalHistory.Answer = NO,
+        hasDiabetes: MedicalHistory.Answer = NO,
+        hasHadKidneyDisease: MedicalHistory.Answer = NO,
+        protocolDrug: ProtocolDrug?,
+        appointmentDate: LocalDate?
+    ) {
+      val patientUuid = patientRepository.saveOngoingEntry(testData.ongoingPatientEntry(fullName = fullName))
+          .andThen(patientRepository.saveOngoingEntryAsPatient())
+          .blockingGet()
+          .uuid
+
+      bpMeasurement?.forEach {
```

Comment:
`BloodPressureRepository` already has a method to save a list of `BloodPressureMeasurement` records. Do you want to use that instead?

Reply:
I need to add `patientUuid` to every `BloodPressureMeasurement`. Since I won't have that value outside this method `savePatientRecord`, I will have to update the `bpMeasurement` for each record to add `patientUuid`.